R/verb-summarise.R

```diff
@@ -53,10 +91,19 @@ op_vars.op_summarise <- function(op) {
 #' @export
 op_grps.op_summarise <- function(op) {
   grps <- op_grps(op$x)
-  if (length(grps) == 1) {
+  .groups <- op$args$.groups %||% "drop_last"
+
+  if (identical(.groups, "drop_last")) {
+    n <- length(grps)
+    if (n == 1) {
```

Comment:
Do you need the if here? I think `grps[-n]` will work even if `grps` is length 1 (or even length 0).

I'd also suggest writing this as a switch:

```R
switch(.groups, 
  drop_last = grps[-length(grps)],
  keep = grps,
  drop = chacracter()
)
```

That makes it more clear that you're dealing with a small set of possible options.

Reply:
It looks like `grps[-n]` works I just didn't use it as currently it is done explicitly like that

```r
if (length(grps) == 1) {
  character()
} else {
  grps[-length(grps)]
}
```

But I don't see a reason why it shouldn't work as `op_grps()` should always return a character.